syntax = "proto3";

import "google/protobuf/any.proto";

message DatabaseRequest {
  string db_name = 1;
  string command = 2;
}

message ErrorStatus {
  string message = 1;
  repeated google.protobuf.Any details = 2;
}

message DatabaseResponse {
  optional ErrorStatus err_status = 1;

  repeated google.protobuf.Any table_column_infos = 2;
  repeated google.protobuf.Any table_rows = 3;
  uint32 row_count = 4;
}

service DatabaseQuery {
  rpc PerformQuery(DatabaseRequest) returns (DatabaseResponse) {}
}
